### PR TITLE
Moved hive macros to the providers package

### LINKED ISSUE #19445: Move Hive macros out of Airflow core to Apache provider
### Body

We have hive macros in Airflow core:
https://github.com/apache/airflow/blob/main/airflow/macros/hive.py

These macros needs to be moved to provider:
https://github.com/apache/airflow/tree/main/airflow/providers/apache/hive

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by ghost**
closes #19445 

This PR contains the following changes:

* A new package `airflow.providers.apache.hive.macros` has been created (cd896e3a20e37f2ca0fca97b84540c7c575e0108)
* The hive macro module previously residing under `airflow.macros.hive` has been moved to `airflow.providers.apache.hive.macros.hive`
* Affected import statments have been modified (2f6d68258fed59c3bc5514f538ee276db987ebcd)

The `airflow.macros` package still remains since there are macros residing in `__init__.py` unrelated to hive.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/macros/__init__.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 import dateutil  # noqa
 
-from airflow.macros import hive  # noqa
+from airflow.providers.apache.hive.macros import hive  # noqa
```
Comment (mik-laj) [inline airflow/macros/__init__.py:25]: We should not import modules from the airflow.providers package in the top-level code as this creates hard dependencies. Now everyone must have Apache Hive provider installed to use the `airflow.macros` module.

We should extend [airflow.providers_manager](https://github.com/apache/airflow/blob/main/airflow/providers_manager.py) with macro support. It should look similar to support macros in plugin_manager.
https://github.com/apache/airflow/blob/3a7e687a369986a35c83ab374ed0e9dc040ecdae/airflow/plugins_manager.py#L428-L458

**Review by potiuk — CHANGES_REQUESTED** [review]
I am afraid this is not enough. Currently macros are loaded by plugins manager: https://github.com/apache/airflow/blob/3a7e687a369986a35c83ab374ed0e9dc040ecdae/airflow/plugins_manager.py#L428

Providers do not currently have the capability to add `macros' - those can only be done via plugins. I think either adding "macros" to plugins or some backwards compatible mechanism so that plugin manager will be able to load macros from installed providers needs to be implemented to complete this issue.

File: tests/macros/test_hive.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import unittest
 from datetime import datetime
 
-from airflow.macros import hive
+from airflow.providers.apache.hive.macros import hive
```
Comment (veera-rapido) [inline tests/macros/test_hive.py:22]: Tests also should co-locate with other tests of this provider not under and macros package

Comment (mrveera) [inline tests/macros/test_hive.py:22]: Tests also should co-locate with other tests of this provider not under and macros package

**Pull request closed without merging**